File: backend/src/main/java/com/nbaliga/entities/Player.java
package com.nbaliga.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
@Entity
public class Player implements Serializable {

    @Id
    private Long id;
    private String firstName;
    private String lastName;
    private String name;
    private Integer height;
    private Integer weight;
    private Integer age;
    private Integer birthMonth;
    private Integer birthDay;
    private Integer birthYear;
    private String currentLeague;
    private Integer currentLeagueId;
    private String currentTeam;
    private Integer currentTeamId;
    private String contractTeam;
    private Integer contractTeamId;
    private String injury;
    private Integer timeInjured;
    private Boolean inactive;
    private Integer positionNumber;
    private Integer experience;
    private String college;
    private String city;
    private String state;
    private Integer greed;
    private Integer happiness;
    private Integer loyalty;
    private Integer winner;
    private Integer insideScoring;
    private Integer jumpShot;
    private Integer freeThrowShot;
    private Integer threePointShot;
    private Integer handling;
    private Integer passing;
    private Integer offensiveRebounding;
    private Integer defensiveRebounding;
    private Integer postDefense;
    private Integer perimeterDefense;
    private Integer stealing;
    private Integer shotBlocking;
    private Integer quickness;
    private Integer strength;
    private Integer stamina;
    private Integer jumping;
    private String pot_inside;
    private String pot_jumpShot;
    private String pot_freeThrow;
    private String pot_3pShot;
    private String pot_handling;
    private String pot_passing;
    private String pot_oRebounding;
    private String pot_dRebounding;
    private String pot_postDefense;
    private String pot_perimeterDefense;
    private String pot_stealing;
    private String pot_blocking;
    private String currentRating;
    private String futureRating;
    private Integer overallRating;
    private Integer overallPotential;
    private Integer championships;
    private Integer playerOfGame;
    private Integer playerOfWeek;
    private Integer playerOfMonth;
    private Integer contract1;
    private Integer contract2;
    private Integer contract3;
    private Integer contract4;
    private Integer contract5;
    private Integer birdYears;

    @OneToMany(mappedBy = "id", fetch = FetchType.LAZY)
    private List<SeasonStats> seasonStats = new ArrayList<>();


}

File: frontend/src/app/features/player/models/player.ts
import { Stats } from '../../league/stats/models/stats';

export interface Player {
  id: number;
  firstName: string;
  lastName: string;
  name: string;
  height: number;
  weight: number;
  age: number;
  birthMonth: number;
  birthDay: number;
  birthYear: number;
  currentLeague: string;
  currentLeagueId: number;
  currentTeam: string;
  currentTeamId: number;
  contractTeam: string;
  contractTeamId: number;
  injury: string;
  timeInjured: number;
  inactive: boolean;
  positionNumber: number;
  experience: number;
  college: string;
  city: string;
  state: string;
  greed: number;
  happiness: number;
  loyalty: number;
  winner: number;
  insideScoring: number;
  jumpShot: number;
  freeThrowShot: number;
  threePointShot: number;
  handling: number;
  passing: number;
  offensiveRebounding: number;
  defensiveRebounding: number;
  postDefense: number;
  perimeterDefense: number;
  stealing: number;
  shotBlocking: number;
  quickness: number;
  strength: number;
  stamina: number;
  jumping: number;
  pot_inside: string;
  pot_jumpShot: string;
  pot_freeThrow: string;
  pot_3pShot: string;
  pot_handling: string;
  pot_passing: string;
  pot_oRebounding: string;
  pot_dRebounding: string;
  pot_postDefense: string;
  pot_perimeterDefense: string;
  pot_stealing: string;
  pot_blocking: string;
  currentRating: string;
  futureRating: string;
  overallRating: number;
  overallPotential: number;
  championships: number;
  playerOfGame: number;
  playerOfWeek: number;
  playerOfMonth: number;
  contract1: number;
  contract2: number;
  contract3: number;
  contract4: number;
  contract5: number;
  birdYears: any;
  rebsPerGame: number;
  assistsPerGame: number;
  pointsPerGame: number;
  contractLength: number;
  playerTs: number;
  playerFgPct: number;
  seasonStats: Stats[];
  overallPhysical: number;
  overallDefense: number;
  overallMental: number;
  overallOffense: number;
  positionName: string;
  playerTeam: string;
  expiring: string;
}

export interface PlayerPageable {
  content: Player[];
  pageable: Pageable;
  last: boolean;
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  sort: Sort2;
  numberOfElements: number;
  first: boolean;
  empty: boolean;
}

export interface Pageable {
  sort: Sort;
  offset: number;
  pageSize: number;
  pageNumber: number;
  paged: boolean;
  unpaged: boolean;
}

export interface Sort {
  empty: boolean;
  sorted: boolean;
  unsorted: boolean;
}

export interface Sort2 {
  empty: boolean;
  sorted: boolean;
  unsorted: boolean;
}
